akka-persistence/src/main/scala/akka/persistence/Eventsourced.scala

```diff
@@ -546,6 +546,7 @@ private[persistence] trait Eventsourced extends Snapshotter with Stash with Stas
           updateLastSequenceNr(p)
           try {
             peekApplyHandler(p.payload)
+            onPersistSuccess(p.payload)
```

Comment:
Ok, so the way I understand this, is that in the finally block I should check if there are other messages persisted and flush those. I will look into that.